client/src/App.tsx

```diff
@@ -25,19 +25,27 @@ import { Payload } from './lib/payload';
 import { SearchIcon } from '@chakra-ui/icons';
 import { search } from './lib/search';
 
+function debounce(f: (v: string) => void, delay: number) {
+  let lastTimeout: number = 0;
+  return (value: string) => {
+    if (lastTimeout) clearTimeout(lastTimeout);
+    lastTimeout = setTimeout(() => {
+      f(value);
+    }, delay);
+  };
+}
+
 const App: React.ElementType = () => {
-  const [value, setValue] = useState<string>('');
   const [payload, setPayload] = useState<Payload | undefined>(undefined);
   const [error, setError] = useState<string | undefined>(undefined);
 
-  const handleInputChange = async (value: string) => {
+  const handleInputChange = useCallback(debounce(async (value: string) => {
     try {
       setPayload(await search(value));
-      setValue(value);
     } catch (error) {
       setError(error instanceof Error ? error.message : 'Unknown Error');
     }
-  };
+  }, 150), []);
 
   const handleExampleClick = (index: number) => {
     handleInputChange(examples[index]);
```

Comment:
We need to be setting the `Input` component value here

Reply:
Fixed this